README.md

```diff
@@ -67,12 +67,15 @@ form](https://coder.com/contact) or send an email to support@coder.com.
 | imagePullPolicy | string | Sets the policy for pulling a container image across all services. | `"Always"` |
 | ingress.additionalAnnotations | list | Deprecated. Please use `ingress.annotations`. | `[]` |
 | ingress.annotations | object | Additional annotations to be used when creating the ingress. These only apply to the Ingress Kubernetes kind. The annotations can be used to specify certificate issuers or other cloud provider-specific integrations. | `{}` |
+| ingress.deployment | object | Options used to configure the Deployment of the default ingress controller. | `{"kind":"Deployment"}` |
+| ingress.deployment.kind | string | How to deploy the built-in ingress. Can be "Deployment" or "DaemonSet". | `"Deployment"` |
 | ingress.enable | bool | If set to true, a Coder compatible ingress kind will be created. You can configure it with `ingress.annotations` below. | `true` |
 | ingress.host | string | The hostname to use for accessing the platform. This can be left blank, and the user can still access the platform from the external IP or a DNS name that resolves to the external IP address. | `""` |
 | ingress.loadBalancerIP | string |  | `""` |
 | ingress.podSecurityPolicyName | string | The name of the pod security policy the built-in ingress controller should abide. It should be noted that the ingress controller requires the `NET_BIND_SERVICE` capability, privilege escalation, and access to privileged ports to successfully deploy. Ignored if `ingress.useDefault` is false. | `""` |
-| ingress.service | object | Options related to the ingress Kubernetes Service object. | `{"annotations":{}}` |
+| ingress.service | object | Options used to configure the LoadBalancer Service used in the default ingress controller. | `{"annotations":{},"externalTrafficPolicy":"Local"}` |
 | ingress.service.annotations | object | Additional annotations to add to the Service object. For example, to make the ingress spawn an internal load balancer: annotations:  cloud.google.com/load-balancer-type: "Internal" | `{}` |
+| ingress.service.externalTrafficPolicy | string | Denotes whether to route external traffic to node-local or cluster-wide endpoints. "Local" routes external traffic directly to the correct node, while "Cluster" evenly distributes traffic across all nodes, but masks source IPs. | `"Local"` |
```

Comment:
Reading the [Kubernetes documentation for external load balancers](https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#preserving-the-client-source-ip), it says this:

> denotes if this Service desires to route external traffic to node-local or cluster-wide endpoints. There are two available options: Cluster (default) and Local. Cluster obscures the client source IP and may cause a second hop to another node, but should have good overall load-spreading. Local preserves the client source IP and avoids a second hop for LoadBalancer and NodePort type services, but risks potentially imbalanced traffic spreading.

I'm unclear though, if the ingress is proxying for us, doesn't it add an X-Forwarded-For header that we can read for the client's IP address? In which case Cluster seems to make more sense, since it means we have good overall load sharing (per the above description)

Reply:
What happens is if the traffic is proxied from a different node before hitting our ingress, we lose the original client IP. There isn't a way to preserve this through kube-proxy unfortunately.

Reply:
I still don't understand why we lose the original client IP. `kube-proxy` is doing iptables stuff, so I get why, if we try to look at the client IP from `net/http`'s RemoteAddr, it won't be correct. But what I'm saying is that, we can have a middleware somewhere extract the original client IP from the X-Forwarded-For header, and then update RemoteAddr? That's what the Chi RealIP middleware is doing: https://github.com/go-chi/chi/blob/f16f8704ccc32f63fbc20eb364c41050cf339860/middleware/realip.go#L14-L38

Reply:
Here's a well written explanation of the issue: https://www.asykim.com/blog/deep-dive-into-kubernetes-external-traffic-policies

`kube-proxy` is only an issue if it forwards traffic *between* nodes

Reply:
Oh, I guess this is because of the way traffic is routed to the _ingress controller_, which does need to see the client's IP address. I got confused and thought this was referring to the traffic routed from the ingress controller to the _cemanager_ service, which does not actually need to see the client IP address, since it can read the `X-Forwarded-For` header added by the ingress controller. Okay, that all makes sense then.

Reply:
Yep! Apologies for the confusion